### PR TITLE
fix(TDOPS-5327): prepare TUI to pnpm and solve some missing deps in packages

### PR DESCRIPTION
**Pull request by smouillour**
**What is the problem this PR is trying to solve?**
Prepare TUI to a potential move to PNPM (or at least fix talend script to be compatible in project side)
+ add some missing dependencies in packages

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

- [ ] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: fork/dynamic-cdn-webpack-plugin/src/find.js [hunk 1]
```diff
@@ -34,14 +34,15 @@ function findPackagesFromScopeFolder(scope, name, scopeFolderPath) {
...
 			}
-			// the scope or package name is not the one we look for
-			// if there is a nested node modules folder, we dive into it for the search
-			const nestedNodeModulesPath = path.join(subFolderPath, 'node_modules');
-			if (fs.existsSync(nestedNodeModulesPath)) {
-				return accu.concat(
-					findPackagesFromNonScopeFolder(scope, name, nestedNodeModulesPath, []),
-				);
-			}
+			// NOT COMPATIBLE WITH PNPM
```
Comment (smouillour) [inline fork/dynamic-cdn-webpack-plugin/src/find.js:37]: This is not compatible with PNPM, it should be ok to go without it

File: packages/design-docs/.storybook/main.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import path from 'path';
+import { packageDirectorySync } from 'pkg-dir';
+
 const iconConfig = require('@talend/icons/.storybook/main.js');
-const rootPath = require.resolve('@talend/design-docs').replace('src/index.ts', '');
```
Comment (smouillour) [inline packages/design-docs/.storybook/main.js:3]: @talend/design-docs should not depends on itself

File: packages/storybook-docs/src/components/Statuses/Statuses.tsx [hunk 3]
```diff
@@ -1,5 +1,4 @@
 import { Unstyled } from '@storybook/blocks';
-import { StackHorizontal, StackItem } from '@talend/design-system';
```
Comment (smouillour) [inline packages/storybook-docs/src/components/Statuses/Statuses.tsx:2]: not used in the file

File: packages/storybook-docs/src/components/Use.tsx [hunk 4]
```diff
@@ -1,41 +1,37 @@
 import type { HTMLAttributes, ReactElement, PropsWithChildren } from 'react';
-// eslint-disable-next-line @talend/import-depth
-import { IconNameWithSize } from '@talend/icons/dist/typeUtils';
```
Comment (smouillour) [inline packages/storybook-docs/src/components/Use.tsx:3]: remove dependency to icons as icon depends on it (cycle deps)

Comment (smouillour) [reply packages/storybook-docs/src/components/Use.tsx:3]: result

File: tools/scripts-config-storybook-lib/.storybook-templates/main.js [hunk 5]
```diff
@@ -26,11 +26,19 @@ const defaultMain = {
...
 		options: {
 			builder: {
-				disableTelemetry: true,
-				enableCrashReports: false,
+				fsCache: true,
+				//lazyCompilation: true
 			},
 		},
 	},
+	typescript: {
```
Comment (smouillour) [inline tools/scripts-config-storybook-lib/.storybook-templates/main.js:34]: improvements for storybook

File: tools/scripts-config-storybook-lib/.storybook-templates/msw/mockServiceWorker.js [hunk 6]
```diff
@@ -2,7 +2,7 @@
 /* tslint:disable */
 
 /**
- * Mock Service Worker (0.49.3).
+ * Mock Service Worker (1.3.2).
```
Comment (smouillour) [inline tools/scripts-config-storybook-lib/.storybook-templates/msw/mockServiceWorker.js:5]: command for msw has been run after upgrade

File: fork/dynamic-cdn-webpack-plugin/src/find.js (hunk 1 above)
Comment (smouillour) [reply fork/dynamic-cdn-webpack-plugin/src/find.js:37]: Note: Not sure of the impact of this change

Comment (jmfrancois) [reply fork/dynamic-cdn-webpack-plugin/src/find.js:37]: This is a fix, we can't merge that. It was the only way to find all versions of a given modules.
For example if you have prop-types twice and one of them can not be hoisted.

File: packages/cmf-cqrs/package.json [hunk 7]
```diff
@@ -33,13 +33,15 @@
   "homepage": "https://github.com/Talend/ui/cmf-cqrs#readme",
   "dependencies": {
     "@talend/react-cmf": "^7.3.0",
-    "@talend/utils": "2.6.0",
```
Comment (jmfrancois) [inline packages/cmf-cqrs/package.json:36]: :o

File: packages/components/package.json [hunk 8]
```diff
@@ -72,10 +73,13 @@
     "@talend/bootstrap-theme": "^8.2.0",
     "@talend/design-system": "^7.15.1",
     "@talend/icons": "^6.60.1",
+    "@talend/eslint-plugin": "^1.0.1",
```
Comment (jmfrancois) [inline packages/components/package.json:76]: this is a dependency of scripts-core so you should not need it as it is not used directly

File: packages/components/package.json [hunk 9]
```diff
@@ -72,10 +73,13 @@
...
     "@talend/locales-design-system": "^7.3.0",
     "@talend/locales-tui-components": "^9.5.0",
+    "@talend/scripts-config-stylelint": "^4.0.0",
```
Comment (jmfrancois) [inline packages/components/package.json:79]: this is not part of core but it should, so not here

File: packages/components/package.json [hunk 10]
```diff
@@ -72,10 +73,13 @@
...
     "@talend/scripts-core": "^15.0.0",
     "@talend/scripts-config-react-webpack": "^16.1.0",
+    "@talend/scripts-config-typescript": "^11.1.0",
```
Comment (jmfrancois) [inline packages/components/package.json:82]: it is already part of scripts core.

**Pull request closed without merging**